File: swig/test/test_bign.py
from bee2py import vp2op, memAlloc, memFree
import bee2py

# begin
buf = memAlloc(128)
privkey = memAlloc(64)
pubkey = memAlloc(128)
hash1 = memAlloc(64)
sig = memAlloc(64+32)
der = memAlloc(512)
count = bee2py.new_sizeTarr(1)

sbuf = "0"*100
params = bee2py.bign_params()
err = bee2py.bignParamsStd(params, "1.2.112.0.2.0.34.101.45.3.1")
print(params.l)
sbuf = bee2py.hexFrom(sbuf, params.q, 32)
print(sbuf)

sbuf = "0"*1000
bee2py.hexTo(privkey, "1F66B5B84B7339674533F0329C74F21834281FED0732429E0C79235FC273E269")
bee2py.bignPubkeyCalc(vp2op(pubkey), params, vp2op(privkey))
sbuf = bee2py.hexFrom(sbuf, pubkey, 64)
print(sbuf)

	# // тест Г.6
	# if (beltHash(hash, beltH(), 13) != ERR_OK)
	# 	return FALSE;
	# if (bignSign2(sig, params, der, count, hash, privkey, 0, 0) 
	# 	!= ERR_OK)
	# 	return FALSE;
bee2py.beltHash(vp2op(hash1), bee2py.beltH(), 13)
sbuf = bee2py.hexFrom(sbuf, hash1, 64)
print(sbuf)

bee2py.sizeTarr_setitem(count, 0, 512)
#bee2py.bignParamsEnc(vp2op(der), count, params)
bee2py.bignOidToDER(vp2op(der), count, "1.2.112.0.2.0.34.101.31.81")
c1 = bee2py.sizeTarr_getitem(count, 0)
print(c1)
sbuf = "0"*1000
sbuf = bee2py.hexFrom(sbuf, der, c1)
print(sbuf)

err = bee2py.bignSign2(vp2op(sig), params, vp2op(der), c1, vp2op(hash1), vp2op(privkey), None, 0)
print(err)
sbuf = "0"*1000
sbuf = bee2py.hexFrom(sbuf, sig, 64+32)
print(sbuf)
err = bee2py.bignVerify(params, vp2op(der), c1, vp2op(hash1), vp2op(sig), vp2op(pubkey))
print(err)

# end
memFree(buf)
memFree(privkey)
memFree(pubkey)
memFree(hash1)
memFree(sig)
memFree(der)
bee2py.delete_sizeTarr(count)

File: swig/test/test_bign.go
package main

import "by.bsu/bee2"
import "fmt"

func main() {
    a := bee2.BeltHash_keep()
    fmt.Println(a)
	cbuf := bee2.MemAlloc(512)

	params := bee2.NewBign_params()
	bee2.BignParamsStd(params,"1.2.112.0.2.0.34.101.45.3.1")
	fmt.Println(params.GetL())
	bee2.HexFrom(bee2.Vp2bp(cbuf), bee2.Op2vp(params.GetQ()), 32)
	sbuf := bee2.Cdata(cbuf, 64)
	fmt.Println(string(sbuf))

	privkey := bee2.MemAlloc(64)
	pubkey := bee2.MemAlloc(128)
	bee2.HexTo(privkey, "1F66B5B84B7339674533F0329C74F21834281FED0732429E0C79235FC273E269")
	bee2.BignPubkeyCalc(bee2.Vp2op(pubkey), params, bee2.Vp2op(privkey))
	bee2.HexFrom(bee2.Vp2bp(cbuf), pubkey, 64)
	sbuf = bee2.Cdata(cbuf, 64)
	fmt.Println(string(sbuf))
	
	hash1 := bee2.MemAlloc(64)
	bee2.BeltHash(bee2.Vp2op(hash1), bee2.Op2vp(bee2.BeltH()), 13)

	count := bee2.New_sizeTarr(1)
	bee2.SizeTarr_setitem(count, 0, 512)
	der := bee2.MemAlloc(512)
	bee2.BignOidToDER(bee2.Vp2op(der), count, "1.2.112.0.2.0.34.101.31.81")
	c1 := bee2.SizeTarr_getitem(count, 0)
	fmt.Println(c1)

	sig := bee2.MemAlloc(64+32)
	err := bee2.BignSign2(bee2.Vp2op(sig), params, bee2.Vp2op(der), c1, 
		bee2.Vp2op(hash1), bee2.Vp2op(privkey), 0, 0)
	fmt.Println(err)
	bee2.HexFrom(bee2.Vp2bp(cbuf), sig, 64+32)
	sbuf = bee2.Cdata(cbuf, 64+32)
	fmt.Println(string(sbuf))

	err = bee2.BignVerify(params, bee2.Vp2op(der), c1, 
		bee2.Vp2op(hash1), bee2.Vp2op(sig), bee2.Vp2op(pubkey))
	fmt.Println(err)

	bee2.MemFree(privkey)
	bee2.MemFree(pubkey)
	bee2.MemFree(hash1)
	bee2.MemFree(sig)
	bee2.MemFree(der)
	bee2.Delete_sizeTarr(count)

}